druid/src/command.rs

```diff
@@ -228,37 +229,96 @@ pub mod sys {
 impl Selector {
     /// A selector that does nothing.
     pub const NOOP: Selector = Selector::new("");
+}
 
+impl<T> Selector<T> {
     /// Create a new `Selector` with the given string.
-    pub const fn new(s: &'static str) -> Selector {
-        Selector(s)
+    pub const fn new(s: &'static str) -> Selector<T> {
+        Selector(s, PhantomData)
+    }
+
+    pub(crate) const fn symbol(self) -> SelectorSymbol {
+        self.0
+    }
+}
+
+impl<T: Any> Selector<T> {
+    pub fn carry(self, object: T) -> Command {
+        Command::new(self, object)
     }
 }
 
 impl Command {
     /// Create a new `Command` with an argument. If you do not need
     /// an argument, `Selector` implements `Into<Command>`.
```

Comment:
The other new docs use the word *payload* which I think is fine. However I think we should be consistent. Let's change all old occurrences of *argument* in relation to commands to *payload*.